lib/json-schema/schema/reader.rb

```diff
@@ -98,6 +98,8 @@ def read_uri(uri)
         else
           raise JSON::Schema::ReadRefused.new(uri.to_s, :uri)
         end
+      rescue OpenURI::HTTPError => error
+        raise error, "#{error} #{uri}"
       end
 
       def read_file(pathname)
```

Comment:
We should also rescue `Errno::ENOENT` for file read errors, so they behave in the same way.